src/components/main/main.tsx

```diff
@@ -301,61 +233,17 @@ export const Main = (props: Props) => {
 			sourcebook = FactoryLogic.createSourcebook();
 			sourcebooks.push(sourcebook);
 		}
-		if (sourcebook) {
-			switch(type) {
-				case 'Ancestry':
-					sourcebook.ancestries.push(element as Ancestry);
-					Collections.sort(sourcebook.ancestries, item => item.name);
-					break;
-				case 'Culture':
-					sourcebook.cultures.push(element as Culture);
-					Collections.sort(sourcebook.cultures, item => item.name);
-					break;
-				case 'Career':
-					sourcebook.careers.push(element as Career);
-					Collections.sort(sourcebook.careers, item => item.name);
-					break;
-				case 'Class':
-					sourcebook.classes.push(element as HeroClass);
-					Collections.sort(sourcebook.classes, item => item.name);
-					break;
-				case 'Complication':
-					sourcebook.complications.push(element as Complication);
-					Collections.sort(sourcebook.complications, item => item.name);
-					break;
-				case 'Domain':
-					sourcebook.domains.push(element as Domain);
-					Collections.sort(sourcebook.domains, item => item.name);
-					break;
-				case 'Kit':
-					sourcebook.kits.push(element as Kit);
-					Collections.sort(sourcebook.kits, item => item.name);
-					break;
-				case 'Perk':
-					sourcebook.perks.push(element as Perk);
-					Collections.sort(sourcebook.perks, item => item.name);
-					break;
-				case 'Title':
-					sourcebook.titles.push(element as Title);
-					Collections.sort(sourcebook.titles, item => item.name);
-					break;
-				case 'Item':
-					sourcebook.items.push(element as Item);
-					Collections.sort(sourcebook.items, item => item.name);
-					break;
-				case 'Monster Group':
-					sourcebook.monsterGroups.push(element as MonsterGroup);
-					Collections.sort(sourcebook.monsterGroups, item => item.name);
-					break;
-			}
-		}
+		const sourcebookKey = getSourcebookKey(kind);
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		sourcebook[sourcebookKey].push(element as any);
```

Comment:
i don't like casting type to `any`, but couldn't work out how else to get this (and other similar consolidations on `sourcebook[sourcebookKey]`) to work with the strong type.

Reply:
Updated this to cast sourcebook `as Record<SourcebookElementsKey, Element[]>`, which is a little tighter than using `any`.